tekton-proposal.md

```diff
@@ -0,0 +1,73 @@
+
+If you want automatic triggered CI/CD or build-test-rollout scenarios, I have good news for you...
+
+### Tekton Pipelines
+
+[`github.com/knative/build-pipeline`](https://github.com/knative/build-pipeline) should continue its move to a new GitHub org, to `github.com/tektoncd/pipeline`.
+
+Its charter and roadmap should continue to focus on building infrastructure to run CI/CD workloads, including triggering, workflow, resources, retries, the works. It should not focus on enabling better serverless deployments specifically, but better continuous deployments to any target generally.
+
+When someone asks, "is there a way to run CI/CD on Kubernetes?" we should be able to say, "yes, because Tekton Pipelines focuses on exactly that."
+
+Until separate Tekton governance is established (weeks, not months), Tekton Pipelines work will continue to be overseen by the Knative Build WG -- largely by Christie (@bobcatfish), as she has been doing a great job leading Pipelines work so far -- though this will likely cause confusion and should be resolved as soon as possible. How we resolve that is TBD, and I think that's outside the scope of this proposal.
+
+## Future Directions
+
+There may come a time in the future when we decide that maintaining Knative Build and Tekton Pipelines as separate efforts is not the best use of engineering resources, and we have in the past discussed the possibility of reimplementing Build as a wrapper around Tekton TaskRuns, or having Tekton Pipelines produce a library that Knative Build could use to avoid duplicated code and maintenance. _Neither of these is currently a plan of record_, and before we pursue these or any other future structural changes to Build, those will be proposed and adopted separately in the future. These are only mentioned here as possible future directions worth noting.
+
+## Alternatives Considered
+
+### Keep Pipelines in Knative
+
+The Knative brand is strong, and staying in Knative has the benefit of inertia; all we have to do is..._nothing_.
+
+Ultimately, however, I don't think this resolves enough confusion. Knative Build and Knative Pipelines _sound_ very similar, but they serve two very different purposes. There's a resource in the Pipelines repo that isn't Build but looks and acts very much like one, and will continue to grow features wholly unrelated to serverless deployments -- which should we direct users to? This is the problem we have today, and doing nothing does nothing to solve it.
+
+Knative's brand is also heavily associated with "serverless" -- serverless on-demand request serving, autoscaling to zero, event consumption and developer experience -- and while this is great for Knative's branding and positioning in the market, if we expand Knative to include CI/CD primitives then we'll have to reframe what Knative means in people's minds, which won't be easy.
```

Comment:
I'd associate the 'deploying source' use case with 'serverless'. That point is omitted.

Reply:
Thanks for pointing that out, reworded.